ARCHITECTURE.md

```diff
@@ -0,0 +1,60 @@
+If you want to familiarize yourself with the code base, you are just in the right place!
+
+See also the [feature guide](./doc/adding_features.md), which walks through specifics of the code base, as well as the [developer guide](./doc/dev.md), which has notes on code styling and testing.
+
+## Bird's Eye View
+
+SkyPortal is an example of a Target and Observation Manager (TOM), which , are designed to support the astronomical community to efficiently work with astronomical data sets, identify sources of interest, and obtain follow-up. It aims to be a sophisticated ``full-stack transient ecosystem'' that integrates capabilities of telescope scheduling, observing optimization, spatial catalogs, astronomical data analysis, and others into a single software stack.
+
+## Front-end
+
+The SkyPortal front-end uses React, a component-based UI library, in conjunction with Redux, a front-end state-management library.
+
+Each component definition typically goes in its own file in `static/js/components`.
+
+SkyPortal bundles all Redux-related code (action types, action creators, reducer) associated with a particular branch of the application state together in a file in the `static/js/ducks` directory ([read more about "ducks" modules here](https://github.com/erikras/ducks-modular-redux)).
+
+SkyPortal hydrates new tabs using the redux state of previous tabs to minimize redundant interactions with the server.
+
+## Back-end
+
+The SkyPortal back-end is built using [Tornado](https://www.tornadoweb.org/en/stable/), a Python web application framework that provides its own I/O event loop for non-blocking sockets, making it ideal for use with websockets.
+
+To handle HTTP requests, we define _request handlers_ that are mapped to API endpoints in the application's configuration (in `skyportal/app_server.py` -- see below). Each SkyPortal request handler is a subclass of
+Handler`(defined in`skyportal/handlers/base.py`), a handler that extends Tornado's base [RequestHandler](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler), handling authentication and providing utility methods for pushing websocket messages to the front-end and returning HTTP responses.
+
+Handlers appear in the directory `skyportal/handlers/api`.
+
+## Database
+
+SkyPortal uses a PostgreSQL database to manage persistent state. The SkyPortal Python backend interacts with the PostgreSQL backend using the
+`SQLAlchemy <http://sqlalchemy.org>`_
+`object relational mapper <https://docs.sqlalchemy.org/en/20/orm/tutorial.html>`_.
+Each database table is represented by a Python class, and each table Column
+is represented by a class attribute.
+
+The models defined in SkyPortal can be found in the directory `skyportal/models`, while those found in baselayer can be found in `baselayer/app/models.py`. When adding a new model, one of the GitHub actions automatically will create the alembic migration file, which should then be committed in the directory `alembic/versions`.
```

Comment:
Is that really how we do it? I feel like the GA is here to test it rather than to generate it, for that we have the docs' migration section right?